File: 0205-isomorphic-strings/0205-isomorphic-strings.cpp
class Solution {
public:
    bool isIsomorphic(string s, string t) {
        set<char> dic_s(s.begin(), s.end());
        set<char> dic_t(t.begin(), t.end());
        set<pair<char, char>> dic_s_t;
        for (int i = 0; i < s.size(); ++i){
            dic_s_t.insert(make_pair(s[i], t[i]));
        }
        
        return dic_s.size() == dic_t.size() && dic_t.size() == dic_s_t.size()? true : false;
    }
};

File: 0205-isomorphic-strings/0205-isomorphic-strings.py
class Solution:
    def isIsomorphic(self, s: str, t: str) -> bool:
        mapping = defaultdict(lambda : '')
        for c1, c2 in zip(s,t):
            if c1 not in mapping and c2 not in mapping.values():
                mapping[c1] = c2
        
        reflect = ""
        for c in s:
            reflect += mapping[c]
            
        return reflect == t
        